# Config file for the combined mission
# Namespaces mirror (optionally) the state for better readability

home_robot:
  key: no_params

reach_detection_hotspot:
  default_outcome: Completed
  goal_pose_topic: /move_base/goal
  base_pose_topic: /base_pose_measured
  roco_controller: MpcTrackLocalPlan
  controller_manager_namespace: /smb_highlevel_controller
  timeout: 10.0
  tolerance_m: 0.10
  tolerance_deg: 10
  nav_goal_service_name: /piloting/get_nav_goal

valve_detection:
  detection_service_name : /piloting/detect_valve
  valve_radius: $(arg valve_radius)
  valve_frame: valve_base
  reference_frame: world
  detect_from_tf: true
  lateral_grasp: false

viewpoint_visitor:
  default_outcome: Completed
  joints_configurations:
    - [ 0.00, 0.26, 3.14, -2.27, 0.00, 0.96, 1.57]
    - [-1.57, 0.26, 3.14, -2.27, 0.00, 0.96, 1.57]
    - [ 1.57, 0.26, 3.14, -2.27, 0.00, 0.96, 1.57]

open_gripper:
  position: 0.0
  max_effort: 100
  tolerance: 0.01
  timeout: 10.0
  gripper_action_name: /my_gen3/arm_robotiq_2f_85_gripper_controller/gripper_cmd

grasp:
  controller_name: $(arg manipulation_controller)
  manager_namespace: /my_gen3
  whitelist: [arm_joint_state_controller, arm_robotiq_2f_85_gripper_controller]
  pose_topic_name: /target_pose
  
close_gripper:
  position: 0.6
  max_effort: 100
  tolerance: 0.01
  timeout: 10.0
  gripper_action_name: /my_gen3/arm_robotiq_2f_85_gripper_controller/gripper_cmd

manipulate_valve:
  controller_name: $(arg manipulation_controller)
  manager_namespace: /my_gen3
  whitelist: [arm_joint_state_controller, arm_robotiq_2f_85_gripper_controller]
  theta_step: 0.50
  theta_desired: 2.5
  dt: 0.1
